ARCHITECTURE.md

```diff
@@ -0,0 +1,3386 @@
+    pub description: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum CreatureCategory {
+    Humanoid,
+    Beast,
+    Undead,
+    Demon,
+    Elemental,
+    Construct,
+    Dragon,
+    Aberration,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CreatureBehavior {
+    /// How aggressive is this creature? (0.0 - 1.0)
+    pub aggression: f32,
+    /// Will it flee when wounded?
+    pub flee_threshold: Option<f32>,
+    /// Preferred combat range
+    pub preferred_range: CombatRange,
+    /// Social behavior
+    pub pack_behavior: bool,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AbilityTemplate {
+    pub id: String,
+    pub name: String,
+    pub cooldown: u32,
+    pub damage_type: Option<DamageType>,
+    pub effects: Vec<StatusEffectType>,
+}
+```
+
+**Template Loading:**
+
+```rust
+// src/bestiary/loader.rs
+
+use std::path::Path;
+use std::fs;
+
+pub struct BestiaryLoader;
+
+impl BestiaryLoader {
+    /// Load all creature templates from a directory
+    pub fn load_from_directory(path: &Path) -> Result<Bestiary, LoadError> {
+        let mut templates = HashMap::new();
+
+        for entry in fs::read_dir(path)? {
+            let entry = entry?;
+            let path = entry.path();
+
+            if path.extension().map_or(false, |e| e == "toml") {
+                let content = fs::read_to_string(&path)?;
+                let template: CreatureTemplate = toml::from_str(&content)?;
+                templates.insert(template.id.clone(), template);
+            }
+        }
+
+        Ok(Bestiary { templates })
+    }
+}
+
+/// Runtime bestiary containing all loaded templates
+#[derive(Debug, Clone)]
+pub struct Bestiary {
+    templates: HashMap<String, CreatureTemplate>,
+}
+
+impl Bestiary {
+    /// Instantiate a creature from a template
+    pub fn spawn(&self, template_id: &str, name: Option<String>) -> Option<Creature> {
+        let template = self.templates.get(template_id)?;
+
+        Some(Creature {
+            id: EntityId::new(),
+            template_id: template_id.to_string(),
+            name: name.unwrap_or_else(|| template.name.clone()),
+            stats: template.base_stats.clone(),
+            status_effects: StatusEffectComponent::default(),
+            current_abilities: template.abilities.iter()
+                .map(|a| ActiveAbility::from_template(a))
+                .collect(),
+        })
+    }
+}
+```
+
+**Example TOML Template:**
+
+```toml
+# data/bestiary/goblin_warrior.toml
+
+id = "goblin_warrior"
+name = "Goblin Warrior"
+category = "Humanoid"
+challenge_rating = 0.25
+description = "A small, green-skinned creature wielding a rusty shortsword. Goblins are cowardly alone but dangerous in groups."
+
+tags = ["goblin", "humanoid", "tribal", "cowardly"]
+
+[base_stats]
+strength = 8
+dexterity = 14
+constitution = 10
+intelligence = 8
+wisdom = 8
+charisma = 6
+max_hp = 7
+current_hp = 7
+max_mana = 0
+current_mana = 0
```

Comment:
use dnd shorthands (str, dex, ...) for this (avoid mana, even hp must be defined like 2d6+3)